### PR TITLE
Use surface for getting value

### PR DESCRIPTION
**Pull request by andizer**
## Summary

This PR can be summarized in the following changelog entry:

* N/A - Change the way we get the canonical by using the surface.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Please test this against the features/indexables-frontend branch in Yoast SEO free
* Have WooCommerce active and add a product
* View the product and verify the canonical is in the WooCommerce schema output.
   * In other words: there should be a Schema node with the type `ItemPage`, with an `@id` that is the same as the `<link rel="canonical>` on the page. 

Related to https://github.com/Yoast/wpseo-woocommerce/pull/439

### CODE REVIEW

**Comment by hansjovis** [comment]
Acceptance test: ✅
